lib/src/notification.dart

```diff
@@ -0,0 +1,389 @@
+        final messageTimestamp = DateTime.fromMillisecondsSinceEpoch(messageTalkjsData['timestamp']);
+        final messageSender = talkjsData['sender'];
+
+        messages.add(
+          Message(
+            messageTalkjsData['message']['text'],
+            messageTimestamp,
+            Person(
+              icon: await _androidIconFromUrl(messageSender['photoUrl']),
+              key: messageSender['id'],
+              name: messageSender['name'],
+            ),
+          ),
+        );
+      }
+
+      styleInformation = MessagingStyleInformation(
+        Person(
+          name: 'me',
+        ),
+        groupConversation: talkjsData['conversation']['participants'].length > 2,
+        messages: messages,
+      );
+    }
+  } else {
+    print("📘 _onFCMBackgroundMessage: data['talkjs'] is NOT String");
+    showId = _nextId;
+    _nextId += 1;
+
+    styleInformation = DefaultStyleInformation(false, false);
+  }
+
+  // We default to not playing sounds, unless a non-empty string is provided
+  final playSound = _androidChannel!.playSound.isNotEmpty;
+  RawResourceAndroidNotificationSound? sound;
+
+  // We use the string 'default' for the default sound (for compatibility with the React Natve SDK)
+  if (playSound && (_androidChannel!.playSound != 'default')) {
+    sound = RawResourceAndroidNotificationSound(_androidChannel!.playSound);
+  }
+
+  final platformChannelSpecifics = NotificationDetails(
+    android: AndroidNotificationDetails(
+      _androidChannel!.channelId,
+      _androidChannel!.channelName,
+      channelDescription: _androidChannel!.channelDescription,
+      importance: _androidChannel!.importance?.toLocalNotification() ?? Importance.high,
+      playSound: playSound,
+      sound: sound,
+      enableVibration: _androidChannel!.vibrate ?? true,
+      vibrationPattern: _androidChannel!.vibrationPattern,
+      channelShowBadge: _androidChannel!.badge ?? true,
+      enableLights: _androidChannel!.lights ?? false,
+      ledColor: _androidChannel!.lightColor,
+      visibility: _androidChannel!.visibility?.toLocalNotification(),
+      styleInformation: styleInformation,
+    ),
+  );
+
+  await _flutterLocalNotificationsPlugin.show(
+    showId,  // id
+    data['title'],  // title
+    data['message'], // body
+    platformChannelSpecifics, // notificationDetails
+    payload: data['talkjs'],
+  );
+}
+
+void _onSelectNotification(NotificationResponse details) {
+  final payload = details.payload;
+
+  print('📘 _onSelectNotification: $payload');
+
+  if (payload != null) {
+    final Map<String, dynamic> talkjsData = json.decode(payload);
+    final String notificationId = talkjsData['conversation']['id'];
+    _activeNotifications.remove(notificationId);
+  }
+}
+
+void _onFCMTokenRefresh(String token) {
+  print('📘 Firebase onTokenRefresh: $token');
+
+  fcmToken = token;
+
+  // TODO: Update the token on the Talkjs server once we have the data layer SDK ready
```

Comment:
```suggestion

```

Reply:
well, if we remove this comment, then we might forget about it when the data layer SDK is ready, but I agree that on an Open Source project, this comment might be confusing to our users.